rotkehlchen/data_import/importers/bitcoin_tax.py

```diff
@@ -0,0 +1,337 @@
+                return  # skip this csv row
+
+            receive_amount = quote_asset_amount
+            receive_event = HistoryEvent(
+                event_identifier=event_identifier,
+                sequence_index=0,
+                timestamp=timestamp,
+                location=location,
+                asset=receive_asset,
+                balance=Balance(
+                    amount=receive_amount,
+                    usd_value=ZERO,
+                ),
+                notes=memo,
+                event_type=event_type,
+                event_subtype=HistoryEventSubType.RECEIVE,
+            )
+            events.append(receive_event)
+            self.add_history_events(cursor, events)
+            return
+
+        spend_event = HistoryEvent(
+            event_identifier=event_identifier,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=location,
+            asset=spend_asset,
+            balance=Balance(
+                amount=spend_amount,
+                usd_value=ZERO,
+            ),
+            notes=memo,
+            event_type=event_type,
+            event_subtype=HistoryEventSubType.SPEND,
+        )
+        receive_event = HistoryEvent(
+            event_identifier=event_identifier,
+            sequence_index=1,
+            timestamp=timestamp,
+            location=location,
+            asset=receive_asset,
+            balance=Balance(
+                amount=receive_amount,
+                usd_value=ZERO,
+            ),
+            notes=memo,
+            event_type=event_type,
+            event_subtype=HistoryEventSubType.RECEIVE,
+        )
+        events.extend([spend_event, receive_event])
+        if fee_amount != ZERO and fee_asset is not None:
+            fee_event = HistoryEvent(
+                event_identifier=event_identifier,
+                sequence_index=2,
+                timestamp=timestamp,
+                location=location,
+                asset=fee_asset,
+                balance=Balance(
+                    amount=fee_amount,
+                    usd_value=ZERO,
+                ),
+                notes=memo,
+                event_type=event_type,
+                event_subtype=HistoryEventSubType.FEE,
+            )
+            events.append(fee_event)
+        self.add_history_events(cursor, events)
+
+    def _consume_income_spending_event(
+            self,
+            cursor: DBCursor,
+            csv_row: dict[str, Any],
+            event_identifier: str,
+            timestamp: TimestampMS,
+            location: Location,
+            action: str,
+            asset: AssetWithOracles,
+            asset_amount: AssetAmount,
+            fee_asset: Optional[AssetWithOracles],
+            fee_amount: Fee,
+            memo: str,
+    ) -> None:
+        try:
+            event_type, event_subtype = self.ACTION_TO_HISTORY_EVENT_TYPE[action]
```

Comment:
Using exceptions in places where the logic is known beforehand is an antipattern. Using `if action not in self.ACTION_TO_HISTORY_EVENT_TYPE: raise ...` will be better here